app/models/user.rb

```diff
@@ -59,7 +59,7 @@ class User < ApplicationRecord
 
   default_value_for :failed_login_attempts, 0
 
-  scope :with_same_userid, ->(id) { where(:userid => User.where(:id => id).pluck(:userid)) }
+  scope :with_same_userid, ->(id) { where(:userid => User.default_scoped.where(:id => id).select(:userid)) }
```

Comment:
wow, this is confusing... so `with_same_userid` is actually "with same user.id` instead of "with same user.userid".  hmm.

Reply:
How much do we use this?  Can we rename it to `with_same_user_id`?  `User` having `userid` column really makes this confusing.

Reply:
no. it is what you expect: "given a `user.id` find all `users` with the same `user.userid`. There are 2 lookups because it uses the `user.id` to lookup a user to determine the `userid`.

It is used in one place https://github.com/ManageIq/manageiq/blob/master/app/models/miq_request.rb#L71
and that is only used in one place https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/miq_request_controller.rb#L342-L343

This can be simplified to accept a user, but since this is a scope used in the ui, I wouldn't be surprised if it was putting that into the session so it passes the `user.id`

Reply:
> "given a user.id..."  

`scope :with_same_userid, ->(id)`

This is the part that's confusing when we have columns called `userid` and `id` on `users`.  

I suggest that it would be clearer that we're passing the `User#id` and not the `User#userid` to the lambda if we call it:

`scope :with_same_user_id, ->(id)`

Reply:
Yes, this is confusing.

While it passes a `user.id` for a record, it wants to match the `userid`

do we want to name the parameter `user_id`?
```ruby
def same_userid(id)
  User.where(:userid => User.find(id).userid)
end
```

Reply:
We got on chat and decided that we should try `User.in_all_regions(id)` as the scope as feels more clear in the client code and hides the ugly details of our username/userid/user.id implementation.

Reply:
Note, it was @kbrock's idea to try `in_all_regions` and he convinced me.  👍